modules/rostests/apitests/user32/ScrollBarRedraw.c

```diff
@@ -0,0 +1,712 @@
+{
+    CurrentColor = (CurrentColor + 1) % TEST_COLOR_COUNT;
+    SetClassLongPtr(Window,
+                    GCLP_HBRBACKGROUND,
+                    (LONG_PTR)ColorBrushes[CurrentColor]);
+
+    trace("New window size: %d x %d, new color: 0x%.8lX\n",
+          NewWidth, NewHeight, Colors[CurrentColor]);
+
+    return 0;
+}
+
+static LRESULT CALLBACK WindowProc(HWND Window, UINT Message, WPARAM wParam, LPARAM lParam)
+{
+    RECT Rect;
+    BOOL BoolRC;
+    int Width;
+    int Height;
+
+    switch (Message)
+    {
+        case WM_CREATE:
+
+            /* It's important for the test that the entire Window is visible. */
+            BoolRC = SetWindowPos(Window, HWND_TOPMOST, 0, 0, 0, 0, SWP_NOMOVE | SWP_NOSIZE);
+            ok(BoolRC, "Failed to set window as top-most, code: %ld\n", GetLastError());
+
+            BoolRC = GetClientRect(Window, &Rect);
+            ok(BoolRC, "Failed to retrieve client area dimensions, code: %ld\n", GetLastError());
+
+            if (BoolRC)
+            {
+                ClientWidth = Rect.right;
+                ClientHeight = Rect.bottom;
+            }
+            else
+            {
+                return -1;
+            }
+
+            BoolRC = GetWindowRect(Window, &Rect);
+            ok(BoolRC, "Failed to retrieve window dimensions, code: %ld\n", GetLastError());
+
+            if (BoolRC)
+            {
+                OrigWidth  = Rect.right - Rect.left;
+                OrigHeight = Rect.bottom - Rect.top;
+            }
+            else
+            {
+                return -1;
+            }
+
+            SmallWidth  = max((OrigWidth  * 3) / 4, 1);
+            SmallHeight = max((OrigHeight * 3) / 4, 1);
+            OrigWidth   = max(OrigWidth,  SmallWidth+1);
+            OrigHeight  = max(OrigHeight, SmallHeight+1);
+
+            trace("OrigWidth: %d, OrigHeight: %d, SmallWidth: %d, SmallHeight: %d\n",
+                  OrigWidth, OrigHeight, SmallWidth, SmallHeight);
+
+            HideVertScrollBar(Window);
+            HideHorzScrollBar(Window);
+
+            FsmState = FSM_STATE_START;
+            FsmTimer = 0;
+
+            return 0;
+
+        case WM_PAINT:
+
+            if (FsmTimer == 0)
+            {
+                FsmTimer = SetTimer(Window, 1, FSM_STEP_PERIOD_MS, NULL);
+                ok(FsmTimer != 0, "Failed to initialize FSM timer, code: %ld\n", GetLastError());
+
+                if (FsmTimer == 0)
+                {
+                    DestroyWindow(Window);
+                    return 0;
+                }
+            }
+            break;
+
+        case WM_SIZE:
+
+            Width = LOWORD(lParam);
+            Height = HIWORD(lParam);
+
+            if (Width != 0 && Height != 0 &&
+                (Width != ClientWidth || Height != ClientHeight))
+            {
+                return OnResize(Window, Width, Height);
+
+                ClientWidth = Width;
+                ClientHeight = Height;
```

Comment:
These two lines are currently dead code since they are _**after**_ the `return` statement; please fix it as you were intended the code to be.

Reply:
Thank you for pointing this out.  I noticed it as well when I was cleaning up the code today, and it helped me uncover quite an interesting bug in my code:
![image](https://github.com/user-attachments/assets/a1f88231-fb9e-4d47-a6cb-6c80da502c65)

It turns out that WM_ERASEBKGND is issued before WM_SIZE.  This means that switching the color in WM_SIZE produces incorrect results, as the previous color was already used.

This means that I think I'll need to switch to using WM_PAINT instead of the implicit WM_ERASEBKGND handler.